src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -273,7 +274,10 @@ public void testExistingTargetFileButWithoutReadAccess()
                 + System.lineSeparator(), systemOut.getLog());
             assertEquals("Unexpected system error log", "", systemErr.getLog());
         });
-        Main.main("-c", "/google_checks.xml", "/etc/sudoers");
+        final Path path = Files.createTempFile("", "");
+        path.toFile().setReadable(false);
+        Main.main("-c", "/google_checks.xml", path.toString());
+        Files.delete(path);
```

Comment:
Wouldn't `Files.delete` have to be wrapped in a try-finally incase Main threw an exception?

Reply:
did slightly differently:
we already have temporary folder Rule in this test class.
I created file under it, so it will be cleaned up with folder after test or tests are finish to execute.